src/webgpu/util/texture/texelData.spec.ts

```diff
@@ -0,0 +1,377 @@
+export const description = 'Test helpers for texel data produce the expected data in the shader';
+
+import { params, poptions } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { unreachable, assert } from '../../../common/framework/util/util.js';
+import {
+  kEncodableTextureFormats,
+  kEncodableTextureFormatInfo,
+  EncodableTextureFormat,
+} from '../../capability_info.js';
+import { GPUTest } from '../../gpu_test.js';
+
+import { getTexelDataRepresentation, TexelComponent } from './texelData.js';
+
+export const g = makeTestGroup(GPUTest);
+
+let ReadbackTypedArray: Float32ArrayConstructor | Int32ArrayConstructor | Uint32ArrayConstructor;
+
+function doTest(
+  t: GPUTest,
+  format: EncodableTextureFormat,
+  componentData: { [C in TexelComponent]?: number }
+) {
+  const rep = getTexelDataRepresentation(format);
+  const texelData = rep.packData(componentData);
+  const texture = t.device.createTexture({
+    format,
+    size: [1, 1, 1],
+    usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.SAMPLED,
+  });
+
+  t.device.defaultQueue.writeTexture(
+    { texture },
+    texelData,
+    {
+      bytesPerRow: texelData.byteLength,
```

Comment:
For Wx1x1 it should be possible to omit this.

Reply:
Types aren't updated to have that yet

Reply:
right.